### ISSUE TITLE
1.4.13 Content on Hover or Focus - applicable to animated flip cards

### ISSUE THREAD

**Issue by detlevhfischer** [w3c/wcag#4071]
You all know these cards that turn once they receive focus. Often they show an image, and hover (and one would hope also keyboard focus) triggers an animation of the card turning over to show text. I guess ESC should revert that change? And what about the hypothetical inverse situation: the card shows text, and hover focus makes it disappear or reveals a decorative image? Would both situations fall under 1.4.13  Content on Hover or Focus and therefore require a mechanism like ESC to restore the default view?

**Comment by patrickhlauke** [comment]
my take would be yes to both

**Comment by patrickhlauke** [comment]
@detlevhfischer do you think we need to clarify this in the understanding or similar?

**Comment by detlevhfischer** [comment]
@patrickhlauke yes, I can try to draft a sentence.

**Comment by fstrr** [comment]
Still waiting on a note for this.

In looking at the [SC's Note 3](https://www.w3.org/WAI/WCAG22/Understanding/content-on-hover-or-focus.html#:~:text=by%20this%20criterion.-,NOTE%203,-This%20criterion%20applies), [which was added in late 2023](https://github.com/w3c/wcag/blame/main/guidelines/sc/21/content-on-hover-or-focus.html#L26), I wonder if it creates a problem:

> This criterion applies to content that appears in addition to the triggering component itself. Since hidden components that are made visible on keyboard focus (such as links used to skip to another part of a page) do not present additional content they are not covered by this criterion.

The flip card is the triggering component, so it can't add more content in addition to itself as it presumably already contains all of its content, it's just that some of it is hidden by default. It does show different content from that of its default view, but wouldn't those be the hidden components that are made visible when the flip card receives keyboard focus?

**Comment by detlevhfischer** [comment]
@fstrr 
One line to take would be to generalise this to include skip links where they indeed cover other content, and let go of the argument added in Note 3 that hidden components made available on keyboard focus do not present additional content. Most of the annoying popovers covering other content are already lurking in hidden state the DOM and should not be exempt - otherwise the entire SC would have no teeth. In practice (for the user) it is irrelevant if content on hover or focus are hidden or generated / inserted. So I think Note 3 was probably a mistake, quite independent of the issue of flip cards.

**Comment by patrickhlauke** [comment]
so would that require that if you have a skip link that appears on focus, you must be able to close/hide that skip link again without moving focus? how would you technically achieve that? and would that skip link that is "focused but now hidden" not contravene focus visible now?

**Comment by detlevhfischer** [comment]
I am not sure - but let me improvise: KB focus makes skip link visible - user chooses whether to activate, whether to use ESC to hide it (presumably focus stays there even when it's not visible, but that was the user's choice - or focus moves to an adjacent component) or user tabs on, which hides the skip link.

My main point was that I don't get why we felt the need to add Note 3, I think it defeats the purpose of the SC.

**Comment by patrickhlauke** [comment]
this would require changes to all websites out there that use those skip links - which i suspect is quite a few. and would be a change with normative implications (even if we just did it in understanding). not sure that would fly easily...

**Comment by detlevhfischer** [comment]
> require changes to all websites out there that use those skip links

Yes, that wouldn't fly. But I would argue Note 3 should then be more specifically focus on skip links - they way it reads now it appears like opening a very wide loophole - anything already on the page but hidden would be exempted. That probably wasn't the intention, but it might be interpreted that way?
